### DISCUSSION TITLE
Rename `random` to `randomInt`

### DISCUSSION THREAD

**Discussion by aleclarson** [radashi-org/radashi#78] [Ideas]
For clarity's sake and also because we'll want to add `randomFloat`.

```ts
export function randomFloat(min = 0, max = 1): number {
  return min + (max - min) * Math.random()
}

export function randomInt(min: number, max: number): number {
  return randomFloat(min, max + 1) | 0
}
```

Let me take this moment to also suggest adding `randomSecureInt` and `randomSecureFloat` functions too, based on the WebCrypto `crypto.getRandomValues` API (available in the global namespace of modern browsers and Node.js v15+).

```ts
declare const crypto: {
  getRandomValues: (array: Uint32Array) => Uint32Array
}

export function randomSecureFloat(min = 0, max = 1): number {
  const int = crypto.getRandomValues(new Uint32Array(1))[0]
  return min + (max - min) * (int / 2 ** 32)
}

export function randomSecureInt(min: number, max: number): number {
  return randomSecureFloat(min, max + 1) | 0
}
```

**Comment by aleclarson** [comment]
Worth asking if `randomDouble` makes more sense, or if it's not a big deal to use float instead. Thoughts?

**Comment by Minhir** [comment]
I'm not sure about the use-cases for `randomSecureX`. In most (all?) scenarios, you want to use it for cryptography and related things. And in such case it's usually not about only one random number but about the big arrays which is more performant to create with native API.  Also, I'm not sure that this is a good idea to trust some secure randomness for 3d-party utility library. 

In the end, `randomSecureX` can be confusing for inexperienced users and useless for experienced ones.

**Reply by aleclarson** [reply]
That's a smart assessment. The [`math-random`](https://www.npmjs.com/package/math-random) package has 1.6M weekly downloads, so it seems there is a need to be served. Though the question is how to do it in a way that's not confusing.

Even if we punt on `randomSecureX` for now, I still plan on renaming `random` and introducing `randomFloat`. ~I also want to switch to a less biased implementation as described in [this article](https://web.archive.org/web/20180728062551/https://pthree.org/2018/06/13/why-the-multiply-and-floor-rng-method-is-biased/).~

**Comment by aleclarson** [comment]
Related issue: https://github.com/sodiray/radash/issues/32
